stonesoup/platform/base.py

```diff
@@ -229,428 +108,58 @@ def remove_sensor(self, sensor: "BaseSensor") -> None:
-            raise AttributeError('transition_models and transition_times must be same length')
-
-        self.transition_index = 0
-        self.current_interval = self.transition_times[0]
+    @velocity.setter
+    def velocity(self, value):
+        self.movement_controller.velocity = value
 
     @property
-    def transition_model(self):
-        return self.transition_models[self.transition_index]
+    def orientation(self):
+        return self.movement_controller.orientation
 
-    def move(self, timestamp=None, **kwargs) -> None:
-        """Propagate the platform position using the :attr:`transition_model`.
+    @orientation.setter
+    def orientation(self, value):
+        self.movement_controller.orientation = value
 
-        Parameters
-        ----------
-        timestamp: :class:`datetime.datetime`, optional
-            A timestamp signifying the end of the maneuver (the default is ``None``)
-
-        Notes
-        -----
-        This methods updates the value of :attr:`position`.
-
-        Any provided ``kwargs`` are forwarded to the :attr:`transition_model`.
-
-        If :attr:`transition_model` or ``timestamp`` is ``None``, the method has
-        no effect, but will return successfully.
-
-        This method updates :attr:`transition_model`, :attr:`transition_index` and
-        :attr:`current_interval`:
-        If the timestamp provided gives a time delta greater than :attr:`current_interval` the
-        :attr:`transition_model` is called for the rest of its corresponding duration, and the move
-        method is called again on the next transition model (by incrementing
-        :attr:`transition_index`) in :attr:`transition_models` with the residue time delta.
-        If the time delta is less than :attr:`current_interval` the :attr:`transition_model` is
-        called for that duration and :attr:`current_interval` is reduced accordingly.
-        """
-        if self.state.timestamp is None:
-            self.state.timestamp = timestamp
-            return
-        try:
-            time_interval = timestamp - self.state.timestamp
-        except TypeError:
-            # TypeError: (timestamp or prior.timestamp) is None
-            return
-
-        temp_state = self.state
-        while time_interval != 0:
-            if time_interval >= self.current_interval:
-                temp_state = State(
-                    state_vector=self.transition_model.function(
-                        state=temp_state,
-                        noise=True,
-                        time_interval=self.current_interval,
-                        **kwargs),
-                    timestamp=timestamp
-                )
-                time_interval -= self.current_interval
-                self.transition_index = (self.transition_index + 1) % len(self.transition_models)
-                self.current_interval = self.transition_times[self.transition_index]
+    def __iter__(self):
+        return self.movement_controller.__iter__()
```

Comment:
How about changing this for `__getitem__`? This would enable access to states, and make the class iterable still.

Reply:
Agreed. Done